single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultSVGWriter.java

```diff
@@ -899,6 +898,71 @@ private void drawFeederInfo(String prefixId, String feederNodeId, List<Point> po
         root.appendChild(g);
     }
 
+    protected void insertBusInfo(String prefixId,
+                                 Point vlShift,
+                                 double shiftX,
+                                 Element root,
+                                 BusNode busNode,
+                                 GraphMetadata metadata,
+                                 DiagramLabelProvider initProvider,
+                                 DiagramStyleProvider styleProvider) {
+        Optional<BusInfo> busInfo = initProvider.getBusInfo(busNode);
+        busInfo.ifPresent(info -> {
+            drawBusInfo(prefixId, busNode, vlShift, shiftX, root, info, styleProvider, metadata);
+            addInfoComponentMetadata(metadata, busInfo.get().getComponentType(), false);
+        });
+    }
+
+    private void drawBusInfo(String prefixId,
+                             BusNode busNode,
+                             Point vlShift,
+                             double shiftX,
+                             Element root,
+                             BusInfo busInfo,
+                             DiagramStyleProvider styleProvider,
+                             GraphMetadata metadata) {
+        Element g = root.getOwnerDocument().createElement(GROUP);
+
+        // Position
+        ComponentSize size = componentLibrary.getSize(busInfo.getComponentType());
+        double vlShiftX = vlShift.getX() + shiftX;
+        if (busInfo.getAnchor() == Side.RIGHT) {
+            vlShiftX = vlShift.getX() + busNode.getPxWidth() - shiftX - size.getWidth();
+        }
+        Point origin = new Point(vlShiftX, vlShift.getY() - size.getHeight() / 2);
+        transformComponent(busNode, origin, g);
+
+        // Styles
+        List<String> styles = styleProvider.getSvgNodeStyles(busNode, componentLibrary, layoutParameters.isShowInternalNodes());
```

Comment:
I saw that sld-busbar-section was added there, which is strange. That's how I came up with the comment that BusInfo should be inside the BusNode svg element. Once it would be inside that element, it will inherit its style, and therefore you can replace this line by `new ArrayList<>()`. And you will need to adjust the shift also!